styles/main.css

```diff
@@ -137,27 +137,30 @@ nav a {
     margin-left: 1em;
     margin-right: 1em;
     display: block;
+    border: 4px solid transparent;
 }
 nav{
     width:100%;
-    margin-left: -2em;
+    margin-left: 2em;
     margin-right: 2em;
-    border-bottom: 1px solid rgba(0,0,0,.5);
+    border-bottom: 1px dotted rgba(0,0,0,.3);
     margin-top: -4px;
+    overflow: auto;
 }
 nav a:hover{
     border: 4px solid rgba(0,0,138,.5);
     
 }
 .back-to-top{
-    float:right;
-    margin-right:-.5px;
-    color:rgba(0,0,0,.5);
-    bottom: 3.5em;
-    position: absolute;
-    
-    
+    text-align: right;
+    margin-right:-.5em;
+    opacity: 0.5;
 }
-.back-to-top a:hover{
-    
+.back-to-top:hover{
+    opacity: 1;
+}
+
+.back-to-top a {
+    position: fixed;
+    bottom: 3.5em;

```

Comment:
Use text-align to right-align inline elements.  No need for float here.
The `.back-to-top` element serves as a container in which the `a` element can be aligned.  Giving `a` a fixed position and only setting the vertical position leaves its horizontal position to the normal flow default (which is at the right side of its right-aligned container).  Using `position: fixed` makes the element positioned relative to the browser viewport.